dlext/Sampler.h

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: MIT
+// This file is part of `hoomd-dlext`, see LICENSE.md
+
+#ifndef SAMPLER_H
+#define SAMPLER_H
+
+#include "hoomd/HalfStepHook.h"
+#include <hoomd/extern/pybind/include/pybind11/pybind11.h>
+#include "dlpack.h"
+
+
+class Sampler : public HalfStepHook
+    {
+    public:
+        //! Constructor
+      Sampler(std::shared_ptr<SystemDefinition> sysdef,
+              pybind11::function python_update);
+
+      virtual void setSystemDefinition(std::shared_ptr<SystemDefinition> sysdef);
+
+        //! Take one timestep forward
+      virtual void update(unsigned int timestep);
+    protected:
+      template<typename T>
+      DLTensor wrap(T* const ptr, const int64_t size2 = 1, const uint64_t offset=0, uint64_t stride1_offset = 0);
```

Comment:
I think I still want to be able to access the DLTensors from python. The main reason would be be to be able to copy and restore snapshots. Maybe this could be achieved with a separate function. I'd have to think about it.

Reply:
I think we solved this.
Ownership of the tensor and the data stays in scope on the C++ side at all times.
So as long as we prevent side effects this should be fine.

Reply:
Yes, the second additional member function is good enough for this.